code/modules/mob/living/say.dm

```diff
@@ -158,6 +158,10 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	var/fullcrit = InFullCritical()
 	if((InCritical() && !fullcrit) || message_mode == MODE_WHISPER)
+		if(fullcrit)
+			var/alertresult = alert(src, "You will be immediately killed by this action. Proceed?",,"Accept", "Decline")
+			if(alertresult == "Decline" || QDELETED(src) || !istype(src) || !src.key)
```

Comment:
Run the critical checks again, otherwise we will have people succumbing by saving the prompt for later lol

Reply:
wait you want me to run an if fullcrit again directly after the prompt

Reply:
wait wait i think i get what you mean, it already does run a if fullcrit a few lines down
here's the entire thing
```
	if((InCritical() && !fullcrit) || message_mode == MODE_WHISPER)
		if(fullcrit)
			var/alertresult = alert(src, "You will be immediately killed by this action. Proceed?",,"Accept", "Decline")
			if(alertresult == "Decline" || QDELETED(src))
				return FALSE
		message_range = 1
		message_mode = MODE_WHISPER
		src.log_talk(message, LOG_WHISPER)
		if(fullcrit)
			var/health_diff = round(-HEALTH_THRESHOLD_DEAD + health)
			// If we cut our message short, abruptly end it with a-..
			var/message_len = length_char(message)
			message = copytext_char(message, 1, health_diff) + "[message_len > health_diff ? "-.." : "..."]"
			message = Ellipsis(message, 10, 1)
			last_words = message
			message_mode = MODE_WHISPER_CRIT
			succumbed = TRUE
	else
		src.log_talk(message, LOG_SAY, forced_by=forced)
```